File: src/web_app/src/modules/core/services/api-clients/user-client.ts
import { httpClient } from '@/modules/core/services/api/http-client';
import type { UserDto } from '@/modules/core/services/api/api.models';
import { ApiUtils } from '@/modules/core/services/api/api.utils';

export class UserClient {
	private static readonly urlBase = 'users';

	public static async getMe(): Promise<UserDto | null> {
		const url = `${this.urlBase}/me/`;

		const request = httpClient.get(url).json<UserDto>();
		const apiResponse = await ApiUtils.createApiResponse(request);

		return apiResponse.isSuccess ? apiResponse.data : null;
    }
}

File: src/bot/src/services/api/clients/user_client.py
from ..http_client import HTTPClient, ApiResponse

from src.common.models import UserDto


class UserClient:
    client = HTTPClient("users")

    @classmethod
    def get_user(cls, **kwargs) -> ApiResponse[UserDto]:
        url = "me/"

        return cls.client.check_session(**kwargs).get(url, UserDto)

    @classmethod
    def signup_user(cls, payload, **kwargs) -> ApiResponse[None]:
        return cls.client.check_session(**kwargs).post("", None, data=payload)

    @classmethod
    def apply_for_role(cls, role: str, **kwargs) -> ApiResponse[None]:
        url = f"apply-role/{role}/"

        return cls.client.check_session(**kwargs).post(url, None)
